statusHandler/processStatusHandler.go

```diff
@@ -0,0 +1,52 @@
+package statusHandler
+
+import (
+	"sync"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+var log = logger.GetOrCreate("statusHandler")
+
+type processStatusHandler struct {
+	mutStatus sync.RWMutex
+	isIdle    bool
```

Comment:
as discussed on Slack, perhaps it would be safer to have a counter here, which will be incremented/decremented on `SetToBusy/SetToIdle`, even though `Create/Process/Commit` operations are "forced" to be  serialized

Reply:
right, will do